src/main/kotlin/rx/lang/kotlin/observables.kt

```diff
@@ -107,3 +107,5 @@ fun <T, R> List<Observable<T>>.zip(zipFunction: (args: List<T>) -> R): Observabl
  * Returns an Observable that emits the items emitted by the source Observable, converted to the specified type.
  */
 inline fun <reified R : Any> Observable<*>.cast(): Observable<R> = cast(R::class.java)
+
+fun<T> observableDefer(x: T) = Observable.defer { Observable.just(x) }
```

Comment:
```java
var x = calculate();

observableDefer(x);
```

When you call `observableDefer`, you already have the value `x` and the calculation is not deferred at all. What you want is something like `Observable.fromCallable(Callable<T>)`:

```java
Observable.fromCallable(() -> {
    System.out.println("Deferred");
    return 1;
});

// when we get here, nothing is printed
```